python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -1209,3 +1213,109 @@ def empty_subset(
         cls, partitions_def: Optional[PartitionsDefinition] = None
     ) -> "DefaultPartitionsSubset":
         return cls()
+
+
+class AllPartitionsSubset(
+    NamedTuple(
+        "_AllPartitionsSubset",
+        [
+            ("partitions_def", PartitionsDefinition),
+            ("instance_queryer", "CachingInstanceQueryer"),
```

Comment:
Even though it means adding an extra attribute, I think it would be clearer to split this out into a `dynamic_partitions_store` and an `evaluation_time`. It considerably narrows down the data that someone trying to understand this needs to think about the implementation accessing.

Reply:
sure, I can break that out, and address Claire's concern about the set of partitions changing over time by enforcing the type of the DynamicPartitionsStore to be CachingInstanceQueryer within `__new__`

Reply:
er well doing that isinstance check makes testing a pain, and also in theory it'd be fine to use a CachingDynamicPartitionsLoader instead (yes I could just check against either of those, but seems like a fairly niche issue to guard against)